### ISSUE TITLE
Support Xcode 14.0+ Swift driver

### ISSUE THREAD

**Issue by polac24** [spotify/XCRemoteCache#153]
**Expected/desired behavior**
Xcode 14 introduced the new swift driver that changes the way swift-frontent commands are invoked. 
```
Swift driver, the component that orchestrates Swift front-end invocations, is now integrated into Xcode’s build system, allowing for fine granular dependencies to other build system tasks and explicit scheduling. (72440175)
```

This change is not yet compatible with the simplest `SWIFT_EXEC` override, so for now disabling the new driver has been applied in #152. However, for a full advantage of swift driver, XCRemoteCache could actively integrate into the Xcode <-> swift-driver communication messages - for now, the approach is still unknown.  

**Relevant integration setup**

- [x] CocoaPods cocoapods-xcremotecache plugin
- [x] Automatic integration using `xcprepare integrate ...`
- [x] Manual integration
- [x] Carthage

**Comment by ainopara** [comment]
Is it possible to change the behavior of swift driver through  custom toolchain?
https://github.com/apple/swift-driver#testing-in-xcode-with-custom-toolchain

**Comment by polac24** [comment]
@ainopara, I don't think so. Even if you change a toolchain, (at least for the non SPM-project) Xcode triggers a step `builtin-SwiftDriver`, which then schedules individual `swift-frontend` steps according to the output of the embedded driver. That said, it is impossible to define a wrapper which would modify the "planning session" done in the driver.

**Comment by CharlieSuP1** [comment]
Just wonder is there any progress?

**Comment by polac24** [comment]
If Xcode invokes `xcswiftc` and then `swift-frontend`, that seems like a cache miss. Does the Xcode's output in the report navigator (CMD+9) shows that there was a cache hit?

To trace the cache miss reason, I would look for logs ([docs](https://github.com/spotify/XCRemoteCache/blob/master/docs/FAQ.md#how-can-i-find-xcremotecache-logs)).

**Comment by grigorye** [comment]
The output says "Disabled remote cache for" indeed, thanks! Will take a look on it.

**Comment by grigorye** [comment]
And I figured it out that it indeed invokes xcswiftc first, I apologise for the confusion.
